docs.feldera.com/docs/pipelines/transactions.md

```diff
@@ -0,0 +1,256 @@
+
+* During a transaction commit, the pipeline computes updates to all views in the
+  program. Depending on the volume of ingested data and the complexity of the
+  views, this process can take a significant amount of time. While the commit is
+  in progress, the pipeline neither ingests new inputs nor produces outputs. To
+  provide visibility, the pipeline periodically (every 10 seconds) reports the
+  progress of the current commit operation in the log.
+
+* Once the commit is complete, the pipeline outputs a batch of updates for every view in the program.
+  These updates are processed by the output connectors, which send them to their associated data sinks.
+
+When the transaction is complete, the pipeline goes back into continuous processing mode. The user can start a new transaction any time.
+
+### Monitoring transaction status
+
+The user can monitor the transaction handling status of the pipeline using the [`/stats`](https://docs.feldera.com/api/retrieve-statistics-e-g-performance-counters-of-a-running-or-paused-pipeline) endpoint. Possible statuses are summarized
+in the following table:
+
+
+| Status                  | Description                                     |
+|-------------------------|-------------------------------------------------|
+| `NoTransaction`         | There is currently no active transaction.       |
+| `TransactionInProgress` | There is an active transaction in progress.     |
+| `CommitInProgress`      | The current transaction is being committed.     |
+
+
+When the status is `TransactionInProgress` or `CommitInProgress`, the `transaction_id` attribute contains the current transaction ID.
+
+<Tabs>
+    <TabItem value="rest" label="REST API">
+    ```shell
+    $ curl -s http://localhost:8080/v0/pipelines/my_pipeline/stats | jq -r '.global_metrics.transaction_status'
+    TransactionInProgress
+
+    $ curl -s http://localhost:8080/v0/pipelines/my_pipeline/stats | jq -r '.global_metrics.transaction_id'
+    1
+    ```
+    </TabItem>
+
+    <TabItem value="python" label="Python SDK">
+    ```python
+    transaction_status = pipeline.transaction_status()
+    transaction_id = pipeline.transaction_id()
+    ```
+    </TabItem>
+
+    <TabItem value="fda" label="fda CLI">
+    ```shell
+    $ fda stats my_pipeline --format json | jq -r '.global_metrics.transaction_status'
+    CommitInProgress
+
+    $ fda stats my_pipeline --format json | jq -r '.global_metrics.transaction_id'
+    1
+    ```
+    </TabItem>
+</Tabs>
+
+## Limitations
+
+* Concurrent transactions are not supported. At most one transaction can run at a time. All inputs ingested by the pipeline while the transaction is active are processed as part of the transaction.
+
+* Checkpointing is disabled during a transaction. A checkpoint initiated during a transaction gets delayed until the transaction has committed.
+
+* Transaction rollback is currently not supported.
```

Comment:
/abort

Reply:
is it planned?

Reply:
Actually neither rollback nor abort are supported. I'll write that. Nope, it's not planned right now. It's probably not too hard to implement (just clear the state of all accumulators), but I'd wait for a use case for it.